language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -435,7 +466,25 @@ private void analyze(URI uri, String content) {
     // Ignore files with parsing error
     analysisResults.failedAnalysisFiles().stream().map(ClientInputFile::getClientObject).forEach(files::remove);
     files.values().forEach(client::publishDiagnostics);
+  }
 
+  private Path findBaseDir(URI uri) {
+    return findBaseDir(workspaceFolders, uri);
+  }
+
+  @VisibleForTesting
+  static Path findBaseDir(List<String> workspaceFolders, URI uri) {
+    Path inputFilePath = Paths.get(uri);
```

Comment:
As a safety net, maybe test if uri starts with file://, else log a warning and return

Reply:
Ok, I added that check in `analyze` instead of here.